test/e2e/specs/block-icons.test.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Internal dependencies
+ */
+import '../support/bootstrap';
+import {
+	newPost,
+	newDesktopBrowserPage,
+	insertBlock,
+	searchForBlock,
+} from '../support/utils';
+import { activatePlugin, deactivatePlugin } from '../support/plugins';
+
+const INSERTER_BUTTON_SELECTOR = '.components-popover__content .editor-block-types-list__item';
+const INSERTER_ICON_SELECTOR = `${ INSERTER_BUTTON_SELECTOR } .editor-block-types-list__item-icon`;
+const INSPECTOR_ICON_SELECTOR = '.edit-post-sidebar .editor-block-icon__colors';
+
+async function getInnerHTML( selector ) {
+	return await page.$eval( selector, ( element ) => element.innerHTML );
+}
+
+async function getBackgroundColor( selector ) {
+	return await page.$eval( selector, ( element ) => {
+		return window.getComputedStyle( element ).backgroundColor;
+	} );
+}
+
+async function getColor( selector ) {
+	return await page.$eval( selector, ( element ) => {
+		return window.getComputedStyle( element ).color;
+	} );
+}
+
+async function getFirstInserterIcon() {
+	return await getInnerHTML( INSERTER_ICON_SELECTOR );
+}
+
+describe( 'Correctly Renders Block Icons on Inserter and Inspector', () => {
+	beforeAll( async () => {
+		await newDesktopBrowserPage();
+		await activatePlugin( 'gutenberg-test-block-icons' );
+		// accept the prompt if the post is "dirty"
+		await page.on( 'dialog', async ( dialog ) => {
+			if ( dialog ) {
+				await dialog.accept();
+			}
+		} );
+	} );
+
+	beforeEach( async () => {
+		await newPost();
+	} );
+
+	afterAll( async () => {
+		await newDesktopBrowserPage();
```

Comment:
Is this necessary? I can’t think of any reason to open a page just before test suite ends.

Reply:
I agree we don't need to open a new page when the test is ending. I followed what we are doing in https://github.com/WordPress/gutenberg/blob/ed8e261a3f1bc89aad3a2da38b38285bf9f7fba9/test/e2e/specs/plugins-api.test.js#L20-L23
I thought we might have a reason to disable plugins in a new page, but I removed this line, and it seems everything is fine.